server/__init__.py

```diff
@@ -154,5 +150,7 @@ def make_connection() -> LobbyConnection:
         )
 
     ctx = ServerContext(make_connection, name="LobbyServer")
+    ping_msg = ctx.protocol_class.encode_message({"command": "ping"})
+
```

Comment:
Could this be defined closer to the usage in `ping_broadcast`?
(Still in the same scope, just so it's easier to understand what `ping_message` is when reading `ping_broadcast`. I guess that would mean moving the definition of `ping_broadcast`.)

Reply:
I guess you are caching here so that it's cached for every protocol (not just `QDataStreamProtocol`)?

Reply:
Yea, if you were to create two `ServerContext` objects, each with a different protocol implementation, then the ping message would get encoded twice, once for each protocol implementation.

That's where I'd like to get to eventually. I'd like to be able to spin up a second `ServerContext` on a different port which uses a better protocol. There are still more refactors that need to happen though for the data to be broadcast over all contexts correctly. Right now, any broadcast would only go out over one context.